CMakeLists.txt

```diff
@@ -445,11 +466,14 @@ target_include_directories(${PROJECT_NAME} PUBLIC $<BUILD_INTERFACE:${CMAKE_CURR
 if (BUILD_TESTING)
     enable_testing()
 
+    ############################################################################
```

Comment:
Isn't there a way to have sub "scripts" for cmake? Like you said this file is getting really large and would be good to break up a bit.

Reply:
yup, the standard way to do this is to have a CMakeLists.txt file per "context". You then "include" the context where appropriate
```
if (BUILD_TESTING)
    include(tests/CMakeLists.txt)
endif()
```
`include` can be thought of as just copy-pasting the contents of the file.

`test/CMakeLists.txt` could then include `tests/unit/CMakeLists.txt`, `tests/fuzz/CMakeLists.txt` and so on.

I'm generally in favor of this, since it cleanly separates things out. The one downside is that everything still operates in the same "scope" so it can be hard to tell where variables are coming from/getting defined
```
/CMakeLists.txt
    test/CMakeLists.txt
        fuzz/CMakeLists.txt
```
Variables in `fuzz/CMakeLists.txt` can be defined in `test/CMakeLists.txt` or `/CMakeLists.txt` which is a little annoying, but I still ultimately consider this a readability improvement.

Reply:
It'll also be a several hundred line diff because I'd be changing the indentation level of all of the test blocks 😬 . So I'd vote for doing that in a separate PR. I'll go ahead and open and issue for it.

Reply:
sounds good. yeah i think we should start splitting this up. it's getting very large 😄

Reply:
#4792